File: codility/lesson3/TapeEquilibrium.java
package com.kiroule.javacodingskills.codility.lesson3;

/**
 * @author Igor Baiborodine
 *
 * @see ../TAPE_EQUILIBRIUM_README.md
 */
public class TapeEquilibrium {

  public int solution(int[] A) {
    for (int i = 1; i < A.length; i++) {
      A[i] = A[i - 1] + A[i];
    }
    int minDiff = Math.abs(A[0] * 2 - A[A.length - 1]);
    //int minDiffIndex = 1;

    for (int i = 1; i < A.length - 1; i++) {
      int diff = Math.abs(A[i] * 2 - A[A.length - 1]);
      if (diff < minDiff) {
        minDiff = diff;
        //minDiffIndex = i + 1;
      }
    }
    //System.out.println("min diff index: " + minDiffIndex);
    return minDiff;
  }

}

File: codility/lesson3/TapeEquilibrium.py
"""Solution for the Tape Equilibrium exercise.

Exercise: ../TAPE_EQUILIBRIUM_README.md
Author:   Igor Baiborodine

"""


def solution(a):
    for i in range(1, len(a)):
        a[i] += a[i - 1]
    min_diff = abs(a[0] * 2 - a[len(a) - 1])

    for i in range(1, len(a) - 1):
        diff = abs(a[i] * 2 - a[len(a) - 1])
        if diff < min_diff:
            min_diff = diff

    return min_diff


assert solution([0, 0]) == 0
assert solution([1, 0]) == 1
assert solution([-1, 0]) == 1
assert solution([1, -1]) == 2
assert solution([1, 1]) == 0

assert solution([1, 1, 1]) == 1
assert solution([1, -1, 1]) == 1
assert solution([1, 2, 3]) == 0

assert solution([3, 1, 2, 4, 3]) == 1
assert solution([1, 1, 10, 10, 1, 10]) == 9
assert solution([1, 1, 10, 10, 1, -10]) == 9
assert solution([1, 1, -10, 10, 1, -10]) == 9
